// @coffee-tree/sejong-auth-delegator/dist/cjs/index.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "LoginRequestDto", {
  enumerable: true,
  get: function () {
    return _LoginRequestDto.default;
  }
});
exports.default = void 0;
exports.sejongAuthDelegator = sejongAuthDelegator;
var _AuthController = _interopRequireDefault(require("./controller/AuthController.cjs"));
var _AuthService = _interopRequireDefault(require("./service/AuthService.cjs"));
var _UserProfileService = _interopRequireDefault(require("./service/UserProfileService.cjs"));
var _LoginRequestDto = _interopRequireDefault(require("./dto/LoginRequestDto.cjs"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function sejongAuthDelegator() {
  const authService = new _AuthService.default();
  const userProfileService = new _UserProfileService.default();
  const authController = new _AuthController.default(authService, userProfileService);
  return {
    async getAuthenticatedJsessionId(loginRequestDto) {
      await authController.authenticate(loginRequestDto);
      return authService.cjsessionId;
    },
    async isAuthenticatedUser(loginRequestDto) {
      const isAuthenticated = await authController.authenticate(loginRequestDto);
      return isAuthenticated;
    },
    async getUserProfile(loginRequestDto) {
      return authController.login(loginRequestDto);
    },
    createLoginRequestDto(userId, password) {
      return new _LoginRequestDto.default(userId, password);
    }
  };
}
var _default = exports.default = sejongAuthDelegator;
