// @coffee-tree/sejong-auth-delegator/dist/cjs/service/UserProfileService.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _ProfileResponseDto = _interopRequireDefault(require("../dto/ProfileResponseDto.cjs"));
var _cheerio = _interopRequireDefault(require("cheerio"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
class UserProfileService {
  constructor() {
    this.profileEndpoint = "http://classic.sejong.ac.kr/userCertStatus.do?menuInfoId=MAIN_02_05";
  }
  async fetchUserProfile(jsessionId) {
    try {
      const profileResponse = await fetch(this.profileEndpoint, {
        method: "GET",
        headers: {
          Cookie: `JSESSIONID=${jsessionId}`
        },
        redirect: "follow"
      });
      const profileHtml = await profileResponse.text();
      return this.parseProfileFromHtml(profileHtml);
    } catch (error) {
      console.error(error);
    }
  }
  parseProfileFromHtml(html) {
    const $ = _cheerio.default.load(html);
    const profileData = $("div.contentWrap li dl dd").map((_, el) => $(el).text().trim()).get();
    const profileResponseDto = new _ProfileResponseDto.default(...profileData);
    return profileResponseDto;
  }
}
var _default = exports.default = UserProfileService;
